### ISSUE TITLE
ERROR: handle_notification_sent - incomplete settings being displayed to users

### ISSUE THREAD

**Issue by mmulrthelp** [microsoft/o365-moodle#1157]
Hi @jamesmcq when users submit an assessment in Moodle

We are seeing the following error on screen

![image](https://user-images.githubusercontent.com/16646247/66194127-c2874b80-e68b-11e9-81a7-265817792a39.png)

for a brief second, this is unsettling to our users, what settings for notifications are incomplete?
Also should it be displayed to screen if debug messages are off?

I've tracked this line down to two parts of the code:
https://github.com/microsoft/moodle-local_o365/blob/master/classes/observers.php
https://github.com/microsoft/moodle-local_o365/blob/master/classes/task/bot.php

Please could you shine a light on what settings it checks for, so we can fix them as they seem to be missing?

They seem to wanting the Microsoft Teams part of the local_o365 plugin to be setup, which we are not using. 

```
        $botappid = get_config('local_o365', 'bot_app_id');
        $botappsecret = get_config('local_o365', 'bot_app_password');
        $notificationendpoint = get_config('local_o365', 'bot_webhook_endpoint');
```

Everything else is working very well, many thanks for your hard work.

**Comment by weilai-irl** [comment]
@jamesmcq, James, I think the debug messages in mtrace() is a bit misleading here, and needs to be updated to be more accurate and less ambiguous.

Basically what it does is, whenever a Moodle notification is sent to a user in Moodle, an observer function in local_o365 tries to check if Teams integration and bot are properly configured, and it will try to send out proactively notifications to the user in office365 if applicable. If any of the requirements to send proactive notifications is not met, it would quit observer function running, and print this message. It is NOT an error, but the correct logic, so it's fair to say the message is misleading and needs update.

Regards,
Lai

**Comment by ponsato** [comment]
If a notification is sent to a user through an ajax web service waiting for a specific response, the mtrace is displayed causing a false error in the service response.

In my opinion, this observer is very aggressive, and can give other conflicts when sending different notifications.

Attach error capture:

![image](https://user-images.githubusercontent.com/3080806/67479101-97c25e80-f65d-11e9-95c0-82fcd19d5aca.png)

**Comment by neeesn** [comment]
We have same issue on some different Moodle sites. Is there an ETA on an update?

**Comment by mmulrthelp** [comment]
@jamesmcq is this to be addressed in a later release, seems like the message is worrying quite a few students.

**Comment by jamesmcq** [comment]
@weilai-irl Thoughts on this one?

**Comment by weilai-irl** [comment]
@jamesmcq I'll create a pull request for this soon.

**Comment by jamesmcq** [comment]
@weilai-irl Thanks!

**Comment by troywilliams** [comment]
Also can you please use debugging() function instead of mtrace() so we can control who sees the messages?

https://github.com/moodle/moodle/blob/master/lib/weblib.php#L3171-L3245

**Comment by weilai-irl** [comment]
@jamesmcq James, I have made a new pull request with the fix. Please review.

**Comment by weilai-irl** [comment]
The issue is resolved as part of 3.7.0.4 release.

https://github.com/microsoft/o365-moodle/releases/tag/v20200128_m37
